src/stream/record_batch.rs

```diff
@@ -75,12 +77,21 @@ where
         record_batch: RecordBatch,
         projection_mask: ProjectionMask,
         full_schema: Arc<Schema>,
+        order: Option<Order>,
     ) -> Self {
+        let offset = if order == Some(Order::Desc) {
```

Comment:
```suggestion
        let offset = if matches!(order, Some(Order::Desc)) {
```
This looks a little weird to me, I think either matching would be better or using the matches! macro